niceplots/utils.py

```diff
@@ -197,6 +197,59 @@ def draggable_legend(axis=None, color_on=True, **kwargs):
         legend[idx].draggable()
 
 
+def label_line_ends(ax, lines=None, labels=None, colors=None, **kwargs):
+    """Place a label just to the right of each line in the axes
+
+    Note: Because the labels are placed outside of the axes, this function works best for plots where all lines end as
+    close to the right edge of the axes as possible. Additionally you need to either use constrained_layout=True
+    (as NicePlots styles do), or call plt.tight_layout() after calling this function.
+
+    Parameters
+    ----------
+    ax : Matplotlib axes
+        axes to label the lines of
+    lines : iterable of matplotlib line objects, optional
+        Lines to label, by default all lines in the axes
+    labels : list of strings, optional
```

Comment:
You can add labels to lines when you call `plot` too right? Just like how it could be done for a legend. Is there a way to interface with that? No worries if it's too much of a hassle, but thought it could be nice

Reply:
Yes, those are the labels this uses by default

Reply:
Well that's embarrassing